tests/transforms/test_tape_expand.py

```diff
@@ -328,3 +329,229 @@ class NonDiffPhaseShift(qml.PhaseShift):
 
         assert new_tape is tape
         spy.assert_not_called()
+
+
```

Comment:
Very nice tests! Very neat usage. I see for the first time what this is good for, I think passing it to the device (even though other UIs are possible with the context manager) is great.

Also, this is a blueprint of how to manipulate the tape expansion at will!

Reply:
agree!